README.md

```diff
@@ -344,6 +345,58 @@ Variables are not restricted to only one assigned type. This is where union type
   // myRealRealAge = true // Not possible since myRealRealAge only accepts a number or a string.
 ```
 
+### Intersection
+
+Intersection types are similar to union types (and similar to the `extend` keyword pattern, which is explained further down), but arguably less common (yet, at least equally useful). Before we get into what an intersection type is, I must mention that in the [TypeScript official documentation about the intersection types](https://www.typescriptlang.org/docs/handbook/advanced-types.html), they are showcased first and *before* the union types, however in this guide I will mention them *after* the union types because they have a bit of a more abstract definition.
+
+To give you an idea of what an intersection type is, think of a mathematical intersection equal to [A ∩ B](https://upload.wikimedia.org/wikipedia/commons/thumb/9/99/Venn0001.svg/220px-Venn0001.svg.png), **but even then, that is not completely right in all cases**. In TypeScript, an intersection type is simply a mix (more commonly referred to as a ***mixin***) between two or more types.
+
+**Note however, that this does not mean that you can freely use intersection types anywhere**. For example, this simple declaration will make the compiler throw an error:
+
+```ts
+let stringAndNumber: string & number = 5;
+```
+
+The reason is because `5` is not assignable to type `string`. And that is because `5` is of type `number`, the types `number` and `string` are **incompatible**.
+
+**That** is the limitation of the intersection types. The intersected types *must* be compatible, by this, I mean that **their properties must not overlap each other**. If this condition is met, **the resulting type will have access to all properties**.
```

Comment:
... must be structurally compatible...
because TS is a structurally typed language

Reply:
Nice idea, let's add some detail about structural typings.